docs/style-guide.rst

```diff
@@ -1284,6 +1284,20 @@ Avoiding Naming Collisions
 This convention is suggested when the desired name collides with that of
 an existing state variable, function, built-in or otherwise reserved name.
 
+Underscore Prefix for Non-external Functions and Variables
+=====================================================
+
+* ``_singleLeadingUnderscore``
+
+This convention is suggested for non-external (``internal`` and ``private``) functions and non-external (``private`` or ``internal``) state variables (state variables without a visibility specified are default ``internal``).
+
+There are many reasons to consider this convention but we note just
+one here. When designing a smart contract the external surface area (functions
+that can be called by any account) is an important consideration. 
+Leading underscores allows you to immediately recognize these decisions while working, but more importantly, if you change a function from non-external to external (including ``public``) and rename it accordingly, this forces you to review every call site while renaming. 
+This can be an important manual check against unintended external functions, a common source of security vulnerabilities (avoid find-replace-all tooling for this change).
```

Comment:
> (avoid find-replace-all tooling for this change)

I'm not sure this is a good overall suggestion. It's perfectly fine to do that as long as you don't blindly commit and actually look at the diffs (as should be the common practice).